packages/wdio-xvfb/src/index.ts

```diff
@@ -0,0 +1,197 @@
+        }
+
+        // Install packages that include xvfb-run
+        await this.installXvfbPackages()
+
+        // Verify xvfb-run is now available
+        try {
+            const { stdout } = await execAsync('which xvfb-run')
+            this.log.info(`Successfully installed xvfb-run at: ${stdout.trim()}`)
+        } catch (error) {
+            this.log.error('Failed to install xvfb-run:', error)
+            throw new Error(
+                "xvfb-run is not available after installation. Please install it manually using your distribution's package manager."
+            )
+        }
+    }
+
+    protected async detectDistribution(): Promise<string> {
+        try {
+            const { stdout } = await execAsync('cat /etc/os-release')
+            if (stdout.includes('ubuntu') || stdout.includes('Ubuntu')) {
+                return 'ubuntu'
+            }
+            if (stdout.includes('debian') || stdout.includes('Debian')) {
+                return 'debian'
+            }
+            if (stdout.includes('fedora') || stdout.includes('Fedora')) {
+                return 'fedora'
+            }
+            if (stdout.includes('centos') || stdout.includes('CentOS')) {
+                return 'centos'
+            }
+            if (stdout.includes('rhel') || stdout.includes('Red Hat')) {
+                return 'rhel'
+            }
+            if (stdout.includes('suse') || stdout.includes('SUSE')) {
+                return 'suse'
+            }
+            if (stdout.includes('arch') || stdout.includes('Arch')) {
+                return 'arch'
+            }
+            if (stdout.includes('alpine') || stdout.includes('Alpine')) {
+                return 'alpine'
+            }
+        } catch {
+            // Fallback detection methods
+            try {
+                await execAsync('which apt-get')
+                return 'debian'
+            } catch {
+                try {
+                    await execAsync('which yum')
+                    return 'rhel'
+                } catch {
+                    try {
+                        await execAsync('which dnf')
+                        return 'fedora'
+                    } catch {
+                        try {
+                            await execAsync('which zypper')
+                            return 'suse'
+                        } catch {
+                            try {
+                                await execAsync('which pacman')
+                                return 'arch'
+                            } catch {
+                                try {
+                                    await execAsync('which apk')
+                                    return 'alpine'
+                                } catch {
+                                    return 'unknown'
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
            // Fallback detection methods
            const dist = await Promise.all([
              execAsync('which apt-get').then(() => 'debian', () => {}),
              execAsync('which yum').then(() => 'rhel', () => {}),
              execAsync('which dnf').then(() => 'fedora', () => {}),
              execAsync('which zypper').then(() => 'suse', () => {}),
              execAsync('which pacman').then(() => 'arch', () => {}),
              execAsync('which apk').then(() => 'alpine', () => {})
            ])
            return dist.unshift() || 'unknown'
        }
```

Reply:
Maybe there is an NPM package for this as well?

Reply:
I'll check this.  Definitely need something better than that pyramid.

Reply:
I think rolling our own is better in this case, couldn't find a recently updated package for this.